lib/slosilo/attr_encrypted.rb

```diff
@@ -24,9 +24,14 @@ module ClassMethods
       def attr_encrypted *a
         options = a.last.is_a?(Hash) ? a.pop : {}
         aad = options[:aad]
+
         # note nil.to_s is "", which is exactly the right thing
         auth_data = aad.respond_to?(:to_proc) ? aad.to_proc : proc{ |_| aad.to_s }
-        raise ":aad proc must take one argument" unless auth_data.arity.abs == 1 # take abs to allow *args arity, -1
+        if Gem::Version.new(RUBY_VERSION) >= Gem::Version.new('3.0.2') and aad.respond_to?(:to_proc)
```

Comment:
```suggestion
        if Gem::Version.new(RUBY_VERSION) >= Gem::Version.new('3.0.0') and aad.respond_to?(:to_proc)
```
For the case we'll keep the if

Reply:
Since its breaking change, I'm removing this condition